### ISSUE TITLE
data_generators.py('get_batch_gt:', AttributeError("'NoneType' object has no attribute 'shape'",))

### ISSUE THREAD

**Issue by msha096** [liuwei16/CSP#16]
Using TensorFlow backend.
num of training samples: 2975
WARNING:tensorflow:From /home/mingzhi/anaconda3/envs/py27/lib/python2.7/site-packages/keras/backend/tensorflow_backend.py:2893: calling l2_normalize (from tensorflow.python.ops.nn_impl) with dim is deprecated and will be removed in a future version.
Instructions for updating:
dim is deprecated, use axis instead
2019-05-07 19:46:42.470193: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2019-05-07 19:46:42.547037: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:964] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2019-05-07 19:46:42.547592: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1411] Found device 0 with properties: 
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.582
pciBusID: 0000:01:00.0
totalMemory: 10.91GiB freeMemory: 7.22GiB
2019-05-07 19:46:42.547603: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1490] Adding visible gpu devices: 0
2019-05-07 19:46:42.729170: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] Device interconnect StreamExecutor with strength 1 edge matrix:
2019-05-07 19:46:42.729195: I tensorflow/core/common_runtime/gpu/gpu_device.cc:977]      0 
2019-05-07 19:46:42.729199: I tensorflow/core/common_runtime/gpu/gpu_device.cc:990] 0:   N 
2019-05-07 19:46:42.729394: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1103] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 6964 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)
load weights from data/models/resnet50_weights_tf_dim_ordering_tf_kernels.h5
WARNING:tensorflow:From /home/mingzhi/anaconda3/envs/py27/lib/python2.7/site-packages/keras/backend/tensorflow_backend.py:1299: calling reduce_mean (from tensorflow.python.ops.math_ops) with keep_dims is deprecated and will be removed in a future version.
Instructions for updating:
keep_dims is deprecated, use keepdims instead
Starting training with lr 0.0002 and alpha 0.999
Epoch 1/150
('get_batch_gt:', AttributeError("'NoneType' object has no attribute 'shape'",))
('get_batch_gt:', AttributeError("'NoneType' object has no attribute 'shape'",))
('get_batch_gt:', AttributeError("'NoneType' object has no attribute 'shape'",))
('get_batch_gt:', AttributeError("'NoneType' object has no attribute 'shape'",))
('get_batch_gt:', AttributeError("'NoneType' object has no attribute 'shape'",))
('get_batch_gt:', AttributeError("'NoneType' object has no attribute 'shape'",))
('get_batch_gt:', AttributeError("'NoneType' object has no attribute 'shape'",))
('get_batch_gt:', AttributeError("'NoneType' object has no attribute 'shape'",))
Exception: need at least one array to concatenate
('epoch', 0, 'takes', 0.0, 'mins')
Epoch 2/150
Exception: 
('epoch', 1, 'takes', 0.0, 'mins')
Epoch 3/150
Exception: 
('epoch', 2, 'takes', 0.0, 'mins')
Epoch 4/150
Exception: 
('epoch', 3, 'takes', 0.0, 'mins')
Epoch 5/150
Exception: 
('epoch', 4, 'takes', 0.0, 'mins')
Epoch 6/150
Exception: 
('epoch', 5, 'takes', 0.0, 'mins')
Epoch 7/150
Exception: 
('epoch', 6, 'takes', 0.0, 'mins')
Epoch 8/150
Exception: 
('epoch', 7, 'takes', 0.0, 'mins')
Epoch 9/150
Exception: 
('epoch', 8, 'takes', 0.0, 'mins')
Epoch 10/150
Exception: 
('epoch', 9, 'takes', 0.0, 'mins')
Epoch 11/150
Exception: 
('epoch', 10, 'takes', 0.0, 'mins')
Epoch 12/150
Exception: 
('epoch', 11, 'takes', 0.0, 'mins')
Epoch 13/150
Exception: 
('epoch', 12, 'takes', 0.0, 'mins')
Epoch 14/150
Exception: 
('epoch', 13, 'takes', 0.0, 'mins')
Epoch 15/150
Exception: 
('epoch', 14, 'takes', 0.0, 'mins')
Epoch 16/150
Exception: 
('epoch', 15, 'takes', 0.0, 'mins')
Epoch 17/150
Exception: 
('epoch', 16, 'takes', 0.0, 'mins')
Epoch 18/150
Exception: 
('epoch', 17, 'takes', 0.0, 'mins')
Epoch 19/150
Exception:

**Comment by soongyuu** [comment]
I also encountered the same problem！

**Comment by soongyuu** [comment]
hi, is the problem solved?

**Comment by msha096** [comment]
> hi, is the problem solved?

change the file path
put the cache file in path_to_csp/data/annotations
put the image file in path_to_csp/data/cityperson/images

and it works for me

**Comment by zm55555** [comment]
> hi, is the problem solved?

does it work to you?
it doesn't to me

**Comment by zxinyang38** [comment]
l meet the same error!if you know how to solve it,please tell me.

**Comment by hppy152** [comment]
change the dataset path,it works for me.I use Caltech,the directory structure  is
  .data/caltech/train_3/annotations_new
 .data/caltech/train_3/images
  .data/caltech/test/annotations_new
 .data/caltech/test/images
